pshtt/pshtt.py

```diff
@@ -651,106 +728,41 @@ def https_check(endpoint):
-
-        # Check for missing SAN.
-        if (
-            (("DNS Subject Alternative Names") in msg) and
-            (("[]") in msg)
-        ):
-            endpoint.https_bad_hostname = True
-
-        # Check for certificate expiration.
-        if (
-            (STORE in msg) and
-            (("FAILED") in msg) and
-            (("certificate has expired") in msg)
-        ):
-            endpoint.https_expired_cert = True
-
-        # Check to see if the cert is self-signed
-        if (
-            (STORE in msg) and
-            (("FAILED") in msg) and
-            (("self signed certificate") in msg)
-        ):
-            endpoint.https_self_signed_cert = True
-
-        # Check to see if there is a bad chain
-
-        # NOTE: If this is the only flag that's set, it's probably
-        # an incomplete chain
-        # If this isnt the only flag that is set, it's might be
-        # because there is another error. More debugging would
-        # need to be done at this point, but not through sslyze
-        # because sslyze doesn't have enough granularity
-
-        if (
-            (STORE in msg) and
-            (("FAILED") in msg) and
-            (
-                (("unable to get local issuer certificate") in msg) or
-                (("self signed certificate") in msg)
-            )
-        ):
-            endpoint.https_bad_chain = True
-
-        # Check for whether the hostname validates.
-        if (
-            (("Hostname Validation") in msg) and
-            (("FAILED") in msg) and
-            (("Certificate does NOT match") in msg)
-        ):
-            endpoint.https_bad_hostname = True
+        logging.exception("{}: Unknown exception examining certificate deployment.".format(endpoint.url))
+        utils.debug("{}: Unknown exception examining certificate deployment: {}".format(endpoint.url, err))
 
     try:
-        endpoint.https_cert_chain_len = len(cert_plugin_result.received_certificate_chain)
+        endpoint.https_cert_chain_len = len(cert_plugin_result.certificate_deployments[0].received_certificate_chain)
         if (
                 endpoint.https_self_signed_cert is False and (
-                    len(cert_plugin_result.received_certificate_chain) < 2
+                    endpoint.https_cert_chain_len < 2
                 )
         ):
             # *** TODO check that it is not a bad hostname and that the root cert is trusted before suggesting that it is an intermediate cert issue.
             endpoint.https_missing_intermediate_cert = True
-            if(cert_plugin_result.verified_certificate_chain is None):
+            if cert_plugin_result.verified_certificate_chain is None:
                 logging.warning("{}: Untrusted certificate chain, probably due to missing intermediate certificate.".format(endpoint.url))
                 utils.debug("{}: Only {} certificates in certificate chain received.".format(endpoint.url, cert_plugin_result.received_certificate_chain.__len__()))
```

Comment:
This needs to use `cert_plugin_result.certificate_deployments[]`. The `if` throws an error as it stands because the `CertificateInfoScanResult` class does not have a `verified_certificate_chain` attribute. The use of `cert_plugin_result.received_certificate_chain.__len__()` would throw a similar error.